gp-nested-forms/gpnf-delay-child-notifications-for-parent-payment.php

```diff
@@ -43,12 +43,18 @@ public function init() {
 
 	public function gpnf_should_send_notification( $should_send_notification, $notification, $context, $parent_form, $nested_form_field, $entry, $child_form ) {
 		if ( $context === 'parent' && $this->is_applicable_form( $parent_form['id'] ) ) {
-			$parent_entry             = GFAPI::get_entry( rgar( $entry, 'gpnf_entry_parent' ) );
-			$should_send_notification = in_array( rgar( $parent_entry, 'payment_status' ), array( 'Paid', 'Active' ), true );
+			$parent_entry   = GFAPI::get_entry( rgar( $entry, 'gpnf_entry_parent' ) );
+			$payment_status = rgar( $parent_entry, 'payment_status', '' );
```

Comment:
Can we simplify the change, like-

```php
$should_send_notification = in_array( rgar( $parent_entry, 'payment_status' ), array( '', 'Paid', 'Active' ), true );
```

Were you able to verify David's concern? "Due to order of events, will there be a point where all parent entries (even those that will eventually have a payment status) come through this code before they have a payment status?"